File: graphql-dgs-codegen-core/src/integTest/java/com/netflix/graphql/dgs/codegen/java/testcases/dataclasses/dataClassWithReservedWord/expected/types/SampleType.java
package com.netflix.graphql.dgs.codegen.java.testcases.dataclasses.dataClassWithReservedWord.expected.types;

import java.lang.Object;
import java.lang.Override;
import java.lang.String;
import java.util.Objects;

public class SampleType {
  private String _return;

  public SampleType() {
  }

  public SampleType(String _return) {
    this._return = _return;
  }

  public String getReturn() {
    return _return;
  }

  public void setReturn(String _return) {
    this._return = _return;
  }

  @Override
  public String toString() {
    return "SampleType{return='" + _return + "'}";
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) return false;
    SampleType that = (SampleType) o;
    return Objects.equals(_return, that._return);
  }

  @Override
  public int hashCode() {
    return Objects.hash(_return);
  }

  public static Builder newBuilder() {
    return new Builder();
  }

  public static class Builder {
    private String _return;

    public SampleType build() {
      SampleType result = new SampleType();
      result._return = this._return;
      return result;
    }

    public Builder _return(String _return) {
      this._return = _return;
      return this;
    }
  }
}

File: graphql-dgs-codegen-core/src/integTest/kotlin/com/netflix/graphql/dgs/codegen/cases/dataClassWithReservedWord/expected/types/SampleType.kt
package com.netflix.graphql.dgs.codegen.cases.dataClassWithReservedWord.expected.types

import com.fasterxml.jackson.`annotation`.JsonIgnoreProperties
import com.fasterxml.jackson.`annotation`.JsonProperty
import com.fasterxml.jackson.`annotation`.JsonTypeInfo
import com.fasterxml.jackson.databind.`annotation`.JsonDeserialize
import com.fasterxml.jackson.databind.`annotation`.JsonPOJOBuilder
import com.netflix.graphql.dgs.codegen.cases.dataClassWithReservedWord.expected.Generated
import java.lang.IllegalStateException
import kotlin.Any
import kotlin.Boolean
import kotlin.Int
import kotlin.String
import kotlin.collections.List
import kotlin.jvm.JvmName

@Generated
@JsonTypeInfo(use = JsonTypeInfo.Id.NONE)
@JsonDeserialize(builder = SampleType.Builder::class)
public class SampleType(
  `return`: () -> String = returnDefault,
) {
  private val __return: () -> String = `return`

  @get:JvmName("getReturn")
  public val `return`: String
    get() = __return.invoke()

  private fun `__$fieldValues`(): List<Any?> = listOf(
      if (__return === returnDefault) returnDefault else `return`,
  )

  override fun equals(other: Any?): Boolean = this === other || (other is SampleType &&
      `__$fieldValues`() == other.`__$fieldValues`())

  override fun hashCode(): Int = `__$fieldValues`().hashCode()

  private fun `__$fieldStrings`(): List<String> = listOfNotNull(
      if (__return === returnDefault) null else "return=" + `return`,
  )

  override fun toString(): String = `__$fieldStrings`().joinToString(prefix = "SampleType(", postfix
      = ")")

  @Generated
  public companion object {
    private val returnDefault: () -> String = 
        { throw IllegalStateException("Field `return` was not requested") }
  }

  @Generated
  @JsonPOJOBuilder
  @JsonIgnoreProperties("__typename")
  public class Builder {
    private var `return`: () -> String = returnDefault

    @JsonProperty("return")
    public fun withReturn(`return`: String): Builder = this.apply {
      this.`return` = { `return` }
    }

    public fun build(): SampleType = SampleType(
      `return` = `return`,
    )
  }
}
